### DISCUSSION TITLE
LS441D feedback

### DISCUSSION THREAD

**Discussion by NeQuissimus** [1000001101000/Debian_on_Buffalo#75] [Show and tell]
@1000001101000 

First of all: Thank you for all the work put into this! :pray: 

I just installed Debian Buster onto my LS441D using this guide and everything **just worked**!
I had a few moments during the initial reboot into the installer when I was worried but the reboot just took its time.

Using the Debian backports, I was even able to set up a ZFS RAIDZ2 pool for my data: https://wiki.debian.org/ZFS#Installation

There seems to be an issue with Samba and ZFS that I am still trying to track down:
As I am pushing my data back onto the NAS via Samba, ZFS's `purge` operation reliably crashes the NAS.
`purge` works fine with `smbd` disabled and `smbd` is just fine when there is no ZFS purge running.
Unfortunately, there are absolutely no logs on what happens, so I may have to try tracing this some other way.
And the whole device is just dead (either networking dies or a kernel crash; no responses to `ping` or `ssh`).
Consider this an FYI, not an issue :)

I am also trying to look more into the RTC issues and am currently building the kernel module for it. I'd love to use `rtcwake` in a cron to shut off the NAS overnight and have it come back in the morning.

Beyond the standard stuff, I set up syslog forwarding into LogStash/ElasticSearch.
Elastic does not provide an ARM version of their Journalbeat, so I may have to investigate there. It may be possible to compile that from source.

So far, so good. I am very happy!

**Comment by 1000001101000** [comment]
Good to hear from you!

Sounds like some interesting experiments you're running! My first instinct about the crash would be to suspect an out of memory isssue, how much swap space do you have?

> I am also trying to look more into the RTC issues and am currently building the kernel module for it. I'd love to use rtcwake in a cron to shut off the NAS overnight and have it come back in the morning.

That reminds me, on my long...long todo list is to try to convince Debian to add the driver for that RTC to their armhf kernel. Let me know if that's something you'd be interested in pursuing.

I spent some time a long time ago trying to make WOL work on this device. I don't think the stock firmware actually supports it but as far as I can see the hardware should be capable of it (the WOL pin of the Ethernet chip is connected to the power button). i sadly didn't take great notes at the time though. I had a hard time confirming whether the device was sleeping vs crashing since I never really succeeded at waking it up. Once thing I did learn is that you typically need to add "wakeup-source" to the device-tree entry of whatever you want to be able to wake up the device. I tried doing that with the power button but I believe I ended up with an error that seemed to indicate that the gpio-controller or the associated interrupt-controller driver wasn't set up for wakeup events.....or something like that, I really wish I'd taken better notes.

I think I may have tried an rtcwake on a device but find I can't remember which device or what I tried. It's an area that could use some investigation since most models should support it from a hardware standpoint.

I took a quick look at the kernel source, I see armada-370-netgear-rn102.dts has entries for waking up from an I2C RTC like the one on the LS441DE. Looking at how it's done for that model might help understand how to make it work for this one.

**Comment by NeQuissimus** [comment]
Update: My crashes have nothing to do with ZFS directly. I think your OOM hunch may be correct. If I have too many concurrent sessions into Samba, I see crashes as well.
I am still fighting Samba trying to find a good configuration.

I did diff Buffalo's kernel release and the upstream kernel. Looks like there is a lot to digest in there, definitely something about clocks and sleeping. I am having a hard time understanding that code, so it's on the back burner for now :)

**EDIT**
Here is a curious tidbit that caused me a slight headache:
If you set `max smbd processes = 3` lower than 3, smbd will not be able to serve up any shares to `smbclient`.

**EDIT 2**
For macOS clients, you actually need `max smbd processes` to be at least 4 and even then it is flaky, so I set it to 5 now.

**Comment by 1000001101000** [comment]
Yeah, the way they implemented their changes is very different from how it eventually got mainlined and is quite different from how it is today. 

What is the default for  ```max smbd processes``` ?

**Reply by NeQuissimus** [reply]
The default is `0`, which means no limit. I think that was running me out of RAM.

**Comment by 1000001101000** [comment]
That makes a lot of sense.

I wonder if there are any recommend configs out there for low power devices. This seems like a good setting to add to the post-installation recommendations but I'm guessing there are more out there too.
